src/services/balancer/contracts/contracts/vault.ts

```diff
@@ -43,11 +47,20 @@ export default class Vault {
 
     if (isWeightedLike(type)) {
       tokenMultiCaller.call('weights', poolAddress, 'getNormalizedWeights', []);
+
+      if (isTradingHaltable(type)) {
+        //TODO: This says Error: no matching function (argument="name", value="getSwapEnabled", code=INVALID_ARGUMENT, version=abi/5.4.0)
+        //tokenMultiCaller.call('swapEnabled', poolAddress, 'getSwapEnabled');
```

Comment:
@EndymionJkb I think this is fixed by importing the InvestmentPool__factory as one of the ABIs available to the tokenMulticaller. I've made the changes here: https://github.com/balancer-labs/frontend-v2/pull/875/commits/0c17b7588701423d2ee2c2f5d00ba2dd6ca6855f